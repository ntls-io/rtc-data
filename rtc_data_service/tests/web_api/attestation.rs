use actix_web::test;

use crate::helpers;

#[actix_rt::test]
async fn auth_service_attestation_ok() {
    attestation_ok("/auth/attest").await;
}

#[actix_rt::test]
async fn data_service_attestation_ok() {
    attestation_ok("/data/attest").await;
}

async fn attestation_ok(uri_path: &str) {
    let app = helpers::init_rtc_service().await;

    let req = test::TestRequest::get().uri(uri_path).to_request();
    let resp = test::call_service(&app, req).await;

    insta::assert_debug_snapshot!(resp);

    let body = test::read_body(resp).await;
    insta::assert_debug_snapshot!(body);
}
