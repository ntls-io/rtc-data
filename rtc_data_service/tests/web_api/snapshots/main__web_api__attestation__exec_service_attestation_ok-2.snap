---
source: rtc_data_service/tests/web_api/attestation.rs
expression: body

---
b"Failed to get quote: SGX_QL_INTERFACE_UNAVAILABLE"
